ooniprobe/Utility/TestUtility.mm

```diff
@@ -72,38 +72,56 @@ + (UIColor*)getColorForTest:(NSString*)testName alpha:(CGFloat)alpha{
-    NSString *path = [NSString stringWithFormat:@"https://orchestrate.ooni.io/api/v1/test-list/urls?country_code=%@", cc];
+    NSURLComponents *components = [[NSURLComponents alloc] init];
+    components.scheme = @"https";
+    components.host = @"orchestrate.ooni.io";
+    components.path = @"/api/v1/test-list/urls";
+    NSURLQueryItem *ccItem = [NSURLQueryItem queryItemWithName:@"country_code" value:cc];
     if ([[SettingsUtility getSitesCategoriesDisabled] count] > 0){
         NSMutableArray *categories = [NSMutableArray arrayWithArray:[SettingsUtility getSitesCategories]];
         [categories removeObjectsInArray:[SettingsUtility getSitesCategoriesDisabled]];
-        path = [NSString stringWithFormat:@"%@&category_codes=%@", path, [categories componentsJoinedByString:@","]];
+        NSURLQueryItem *categoriesItem = [NSURLQueryItem queryItemWithName:@"category_codes" value:[categories componentsJoinedByString:@","]];
+        components.queryItems = @[ ccItem, categoriesItem ];
     }
-    NSURL *url = [NSURL URLWithString:path];
-    NSURLSessionDataTask *downloadTask = [[NSURLSession sharedSession] dataTaskWithURL:url completionHandler:^(NSData *data, NSURLResponse *response, NSError *error) {
-        if (!error) {
-            NSDictionary *dic = [NSJSONSerialization JSONObjectWithData:data options:0 error:&error];
-            NSArray *urlsArray = [dic objectForKey:@"results"];
-            NSMutableArray *urls = [[NSMutableArray alloc] init];
-            for (NSDictionary* current in urlsArray){
-                //List for database
-                Url *url = [Url checkExistingUrl:[current objectForKey:@"url"] categoryCode:[current objectForKey:@"category_code"] countryCode:[current objectForKey:@"country_code"]];
-                //List for mk
-                [urls addObject:url.url];
-            }
-            completion(urls);
-        }
-        else {
-            // Fail
-            completion(nil);
-            NSLog(@"error : %@", error.description);
-        }
+    else {
+        components.queryItems = @[ ccItem ];
+    }
+    NSURL *url = components.URL;
+    NSURLSessionDataTask *downloadTask = [[NSURLSession sharedSession]
+     dataTaskWithURL:url
+     completionHandler:^(NSData *data, NSURLResponse *response, NSError *error) {
+         if (error != nil) {
+             errorcb(error);
+             return;
+         }
+         NSDictionary *dic = [NSJSONSerialization JSONObjectWithData:data options:0 error:&error];
+         if (error != nil) {
+             errorcb(error);
+             return;
+         }
+         NSArray *urlsArray = [dic objectForKey:@"results"];
+         NSMutableArray *urls = [[NSMutableArray alloc] init];
+         for (NSDictionary* current in urlsArray){
+             //List for database
+             Url *url = [Url checkExistingUrl:[current objectForKey:@"url"] categoryCode:[current objectForKey:@"category_code"] countryCode:[current objectForKey:@"country_code"]];
```

Comment:
Please, wrap this long line

Reply:
[sudo please wrap this long line](https://xkcd.com/149/)?

Reply:
This line has to be removed as there is need to check for url country_code and category_code
https://github.com/ooni/probe-ios/pull/293#discussion_r298369642

Reply:
Gotcha!